### DISCUSSION TITLE
Parsing prefixes

### DISCUSSION THREAD

**Discussion by TimMeuser** [eclipse-rdf4j/rdf4j#3318] [Q&A]
Hi,
I have two questions:
1. What is the best way to parse SPARQL query prefixes, like PREFIX ex: <http://example.org/> ?
2. Is there a definition of the used rdf4j grammar?

Thanks in advance,
Tim

**Comment by barthanssens** [comment]
Hi Tim,

could you tell us a bit more what you want to achieve ? E.g do you want to write a SPARQL parser yourself, or just turning namespace prefixes into namespaces URI ?

Best regards

Bart
